Feature/invalid format 36637991

Bukarinov opened the pull request:
https://www.pivotaltracker.com/story/show/36637991

shumkov committed:
rename exception

shumkov committed:
start

Bukarinov committed:
ApiStrategy refactoring

Bukarinov committed:
protected etParamsFromRequest

Bukarinov committed:
Oops!

shumkov commented on library/GeometriaLab/Api/Mvc/View/Strategy/ApiStrategy.php:
```diff
@@ -2,128 +2,265 @@
-                $headers->addHeaderLine('content-type', 'application/javascript');
-            } else {
-                $headers->addHeaderLine('content-type', 'application/json');
+            if (is_string($contentType)) {
```
Надо если не коллэбл вызывать в противном случае передавать если там что-то другое то приведет к стринге просто в итоге или уже закричит addHeaderLine

shumkov commented on library/GeometriaLab/Api/Mvc/View/Strategy/ApiStrategy.php:
```diff
@@ -2,128 +2,265 @@
-                // We don't have a string, and thus, no JSON
-                return;
-            }
+        if (!$this->hasRenderer($format)) {
```
Тут уже по идеи проверка не нужна. Кстати не очень правильно что контент тайп не связан с рендером и может не быть. Я бы сетил его вместе с ренедром обязательным образом.

shumkov commented on library/GeometriaLab/Api/Mvc/View/Strategy/ApiStrategy.php:
```diff
@@ -2,128 +2,265 @@
+     * @param \Zend\EventManager\EventManagerInterface $events
+     * @param int $priority
+     */
+    public function attach(ZendEventManagerInterface $events, $priority = 1)
```
Надо добавить событие которые перед деспатчем получает конфигурацию, сетит ее и проверяет формат и выкидывет исключение если он не верный

Bukarinov committed:
content_type is required and must be string or callable

Bukarinov committed:
ApiStrategy refactoring

Bukarinov committed:
Listeners refactoring.

Merge branch 'master' of github.com:GeometriaLab/ZF2Library into feature/invalidFormat-36637991

Conflicts:
	library/GeometriaLab/Api/Mvc/Controller/Action/Params/Listener.php

Bukarinov committed:
Fix params test

Bukarinov committed:
Listeners refactoring

Bukarinov committed:
Listeners refactoring
